src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
```

Comment:
Consider using Object.setPrototypeOf() instead of modifying the Array prototype directly. This will help to avoid conflicts with other code.